dsp/window/window_complex_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package window
+
+import (
+	"math"
+	"testing"
+	"unsafe"
+)
+
+type testingCaseCmplx struct {
+	Name           string
+	Function       func([]complex128) []complex128
+	ExpectedResult []float64
+}
+
+type testingCaseCmplxGauss struct {
+	Name           string
+	Sigma          float64
+	ExpectedResult []float64
+}
+
+func equalApprox(result []complex128, expectedResult []float64, tol float64) bool {
+	if len(result) != len(expectedResult) {
+		return false
+	}
+	for i := range result {
+		if math.Abs(real(result[i])-expectedResult[i]) > tol {
+			return false
+		}
+		if math.Abs(imag(result[i])-expectedResult[i]) > tol {
+			return false
+		}
+	}
+	return true
+}
+
+func TestWindowComplex(t *testing.T) {
+	//test precission
+	const tol = 1e-6
+	//Input data
+	src := make([]complex128, 20)
+	for i := range src {
+		src[i] = complex(1.0, 1.0)
+	}
+
+	//testing table (except gauss window)
+	var tt []*testingCaseCmplx = []*testingCaseCmplx{
```

Comment:
Please place test tables outside the test function and declare them as unnamed types.

```
var complexWindowTests = []struct{
	name string
	fn   func([]complex128) []complex128
	want []complex128
} {
	{
		name: "RectangleComplex", fn: RectangleComplex,
		want: []complex128{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1},
	},
...
```

and

```
var complexGausWindowTests = []struct{
	name  string
	fn    func([]complex128) []complex128
	sigma float64
	want  []complex28
} {
	{
		name: "GaussComplex", fn: GausComplex, sigma: 0.3,
		want: []complex128{0.003866, 0.011708, 0.031348, 0.074214, 0.155344, 0.287499, 0.470444, 0.680632, 0.870660, 0.984728, 0.984728, 0.870660, 0.680632, 0.470444, 0.287499, 0.155344, 0.074214, 0.031348, 0.011708, 0.003866},
	},
...
```

Then split the general tests from the Gaussian tests.